tests/spec/core/biblio-spec.js

```diff
@@ -92,26 +117,104 @@ describe("W3C — Bibliographic References", () => {
     expect(ref.textContent).toMatch(/Publishers Inc\.\s/);
     ref = null;
     // Make sure the ". " is automatically added to publisher.
-    ref = doc.querySelector("#bib-TestRef2 + dd");
+    ref = docDefault.querySelector("#bib-TestRef2 + dd");
     expect(ref).toBeTruthy();
     expect(ref.textContent).toMatch(/Testing 123\.\s/);
     ref = null;
     // Make sure publisher is shown even when there is no author
-    ref = doc.querySelector("#bib-TestRef3 + dd");
+    ref = docDefault.querySelector("#bib-TestRef3 + dd");
     expect(ref).toBeTruthy();
     expect(ref.textContent).toMatch(/Publisher Here\.\s/);
   });
 
+  it("displays the reference for APA citation", () => {
+    // Make sure the reference is added.
+    let ref = docAPA.querySelector("#bib-TestRef1 + dd");
+    expect(ref).toBeTruthy();
+    expect(ref.textContent).toMatch(/William Shakespeare\.\s/);
+    expect(ref.textContent).toMatch(/Test ref title\.\s/);
+    expect(ref.textContent).toMatch("Retrieved from URL: http://test.com/");
```

Comment:
Ok, but now we want to test the finished product:
```JS
const finalString = "William Shakespeare. Test ref title. Retrieved from URL: http://test.com/";
expect(ref.textContent.trim()).toEqual(finalString); 
```

The same applies to each below.

Reply:
We also want to make sure that the right things are hyperlinked:

```JS
const anchors = ref.querySelectorAll("a");
expect(anchors.length).toEqual(2);
const [title, retrievedFrom] = [...anchors];
expect(title.href).toEqual("http://test.com/");
expect(title.firstElementChild.localName).toEqual("cite");
expect(title.firstElementChild.textContent).toEqual("Test ref title");
expect(retrievedFrom.href).toEqual("http://test.com/");
```

And so on... have a look at the code that created this, and make sure the HTML structure is what you expect it to be.